test/e2e/pvc_test.go

```diff
@@ -0,0 +1,255 @@
+package e2e_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	operatorapi "github.com/openshift/api/operator/v1"
+
+	imageregistryv1 "github.com/openshift/cluster-image-registry-operator/pkg/apis/imageregistry/v1"
+	"github.com/openshift/cluster-image-registry-operator/test/framework"
+)
+
+func testDefer(t *testing.T, client *framework.Clientset) {
+	if t.Failed() {
+		scList, err := client.StorageClasses().List(metav1.ListOptions{})
+		if err != nil {
+			t.Logf("unable to dump the storage classes: %s", err)
+		} else {
+			framework.DumpYAML(t, "storageclasses", scList)
+		}
+
+		pvList, err := client.PersistentVolumes().List(metav1.ListOptions{})
+		if err != nil {
+			t.Logf("unable to dump the persistent volumes: %s", err)
+		} else {
+			framework.DumpYAML(t, "persistentvolumes", pvList)
+		}
+
+		pvcList, err := client.PersistentVolumeClaims(imageregistryv1.ImageRegistryOperatorNamespace).List(metav1.ListOptions{})
+		if err != nil {
+			t.Logf("unable to dump the persistent volume claims: %s", err)
+		} else {
+			framework.DumpYAML(t, "persistentvolumeclaims", pvcList)
+		}
+	}
+	framework.MustRemoveImageRegistry(t, client)
+}
+
+func createPV(t *testing.T, storageClass string) error {
+	client := framework.MustNewClientset(t, nil)
+	name := ""
+
+	for i := 0; i < 100; i++ {
+		pvName := fmt.Sprintf("pv-%s", rand.String(64))
+		localPath := fmt.Sprintf("/tmp/%s", pvName)
+
+		if err := os.MkdirAll(localPath, 0777); err != nil {
```

Comment:
this code(the test code) isn't running on the cluster nodes, so how does this work?  I think you're creating PVs that aren't actually backed by real directories on the cluster nodes (which i guess is fine since as you noted, these tests don't push/pull to the registry anyway right now).

Reply:
@bparees Tests pass. It works. PV change status.phase into the `available` state. OK, will remove `mkdir`.

Reply:
I assume this is a callback to our discussion from yesterday, so i'll just say:

If you would like to add a basic image push/pull to our operator suite to ensure the various storage setups work, that would be great.  If not, when we get "e2e-xxxx" where xxxx defaults to PV storage, we will find out whether this works end to end, but assuming the PVC is being created and a PV is being claimed, it I guess I would expect that it is, barring some sort of permissions issue writing to the PV or incorrect configuration setting in the registry pod.

Reply:
>  If not, when we get "e2e-xxxx" where xxxx defaults to PV storage, we will find out whether this works end to end, but assuming the PVC is being created and a PV is being claimed, it I guess I would expect that it is, barring some sort of permissions issue writing to the PV or incorrect configuration setting in the registry pod.

@bparees I dumped all PVs / PVCs  and did not see any other than those created by me. The deployment cannot change status to `processing=false` if the PVC failed to bound PV.